### PR TITLE
test: migrate client module bookie tests to junit 5

### LINKED ISSUE #4322: Migrate tests from junit4 to junit5
JUnit 5 offers significant improvements over JUnit 4, including more powerful and flexible testing capabilities like annotation-based coding(so we don't need to write assume), parallel test execution, and other features. We can write new test classes using junit5, and existing-test classes can be slowly migrated.

This issue to track tests migrate to junit5, we can migrate a package、a small module gradually.

See also:
 https://lists.apache.org/thread/hlyc722gy45s3y61q7n79hdcr6cgrx7v

### PR DESCRIPTION
**Pull request by sherlock-lin**
Descriptions of the changes in this PR:

Fix #4322

Main Issue: #4322

### CODE REVIEW

**Comment by sherlock-lin** [comment]
@shoothzj

**Review by hezhangjian — COMMENTED** [review]
Looks good, I have two advice:
- I think `migrate` might be better than `Upgrade`, WDYT?
- As we have many tests, we should point on which part of tests are migrated in PR title. It's better to do like `Migrate All client tests to junit5`, or `Migrate xxxx module tests to junit5`

**Comment by sherlock-lin** [comment]
> Looks good, I have two advice:
> 
> * I think `migrate` might be better than `Upgrade`, WDYT?
> * As we have many tests, we should point on which part of tests are migrated in PR title. It's better to do like `Migrate All client tests to junit5`, or `Migrate xxxx module tests to junit5`

That's good advice, I'll make the adjustments！

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieRecoveryUseIOThreadTest.java [hunk 1]
```diff
@@ -35,6 +37,7 @@ public BookieRecoveryUseIOThreadTest() {
         super(1);
     }
 
+    @BeforeEach
```
Comment (shoothzj) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieRecoveryUseIOThreadTest.java:40]: I think we don't need this annotation, it's already on the base `BookKeeperClusterTestCase`

**Comment by shoothzj** [comment]
Thanks for your contribution, please fix the checkstyle violations. :)

**Review by hezhangjian — APPROVED** [review]

**Comment by sherlock-lin** [comment]
> Thanks for your contribution, please fix the checkstyle violations. :)

ok

**Review by dlg99 — APPROVED** [review]

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieDecommissionTest.java [hunk 2]
```diff
@@ -35,7 +35,8 @@
 import org.apache.bookkeeper.meta.ZkLedgerUnderreplicationManager;
 import org.apache.bookkeeper.net.BookieId;
 import org.apache.bookkeeper.test.BookKeeperClusterTestCase;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
+
```
Comment (shoothzj) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieDecommissionTest.java:39]: nit: unnecessary blank line

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieNetworkAddressChangeTest.java [hunk 3]
```diff
@@ -20,8 +20,9 @@
  */
 package org.apache.bookkeeper.client;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
+
```
Comment (shoothzj) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieNetworkAddressChangeTest.java:23]: nit: unnecessary blank line

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieRecoveryTest.java [hunk 4]
```diff
@@ -61,6 +61,7 @@ public class BookieRecoveryTest extends BookKeeperClusterTestCase {
 
     // Object used for synchronizing async method calls
     class SyncObject {
+
```
Comment (shoothzj) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieRecoveryTest.java:64]: nit: unnecessary blank line

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieRecoveryUseIOThreadTest.java [hunk 5]
```diff
@@ -18,13 +18,15 @@
...
+
 import java.io.IOException;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.atomic.AtomicInteger;
 import org.apache.bookkeeper.conf.ClientConfiguration;
 import org.apache.bookkeeper.test.BookKeeperClusterTestCase;
-import org.junit.Assert;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
+
```
Comment (shoothzj) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieRecoveryUseIOThreadTest.java:29]: nit: unnecessary blank line

**Comment by sherlock-lin** [comment]
> Thanks for your contribution, and welcome to the community. :)

Glad to join the community. 😃

**Pull request merged**